src/sage/topology/simplicial_complex.py

```diff
@@ -4802,7 +4810,110 @@ def intersection(self, other):
             F = F + [s for s in self.faces()[k] if s in other.faces()[k]]
         return SimplicialComplex(F)
 
+    def bigraded_betti_numbers(self, base_ring=ZZ):
+        r"""
+        Return a dictionary of the bigraded Betti numbers of ``self``, 
+        with keys `(-i, 2j)`.
+
+        .. SEEALSO::
+
+            See :meth:`bigraded_betti_number` for more information.
+
+        EXAMPLES::
+
+            sage: X = SimplicialComplex([[0,1],[1,2],[1,3],[2,3]])
+            sage: Y = SimplicialComplex([[1,2,3],[1,2,4],[3,5],[4,5]])
+            sage: X.bigraded_betti_numbers()
+            {(0, 0): 1, (-1, 4): 2, (-1, 6): 1, (-2, 6): 1, (-2, 8): 1}
+            sage: Y.bigraded_betti_numbers(base_ring=QQ)
+            {(0, 0): 1, (-1, 4): 3, (-2, 6): 1, (-2, 8): 2, (-3, 10): 1}
+        """
+        if self._bbn is not None and 0 in self._bbn and base_ring in self._bbn[0]:
+            return self._bbn[base_ring]
+
+        from sage.homology.homology_group import HomologyGroup
+        L = self.vertices()
+        n = len(L)
+        B = {}
+        H0 = HomologyGroup(0, base_ring)
+
+        B[(0,0)] = ZZ.one()
+
+        for j in range(n+1):
+            for x in combinations(L, j):
+                S = self.generated_subcomplex(x)
+                H = S.homology(base_ring=base_ring)
+                for k in range(j):
+                    if j-k-1 in H and H[j-k-1] != H0:
+                        ind = (-k, 2*j)
+                        if ind not in B:
+                            B[ind] = ZZ.zero()
+                        B[ind] += len(H[j-k-1].gens())
+
+        if self._bbn is not None:
+            if 0 in self._bbn:
+                self._bbn[0].add(base_ring)
+            else:
+                self._bbn[0] = {base_ring}
+            self._bbn[base_ring] = B
+        else:
+            self._bbn = {0: {base_ring}, base_ring: B}
+        return B 
+
+    def bigraded_betti_number(self, a, b, base_ring=ZZ):
+        r"""
+        Return the bigraded Betti number indexed in the form `(-i, 2j)`.
+
+        Bigraded Betti number with indices `(-i, 2j)` is defined as a sum of ranks
+        of `(i-j-1)`-th (co)homologies of full subcomplexes with exactly `j` vertices. 
```

Comment:
```suggestion
        of `(j-i-1)`-th (co)homologies of full subcomplexes with exactly `j` vertices. 
```
Is this a typo? This is what the code below is doing.

Reply:
Yes, it is a mistake. Thank you for noticing.